tests/visualization/positioning_test.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+import torch
+
+from adam.visualization.positioning import (
+    AxisAlignedBoundingBox,
+    main,
+    CollisionPenalty,
+    DistanceFromOriginPenalty,
+)
+
+
+def test_positioning() -> None:
+    np.random.seed(127)
+    main()
```

Comment:
I would split this into three separate tests so that each test tests one thing.

Seeing `main` in a test looks weird to the reader, so perhaps rename it to something else which makes clear what it is doing.  Also comment that we are just checking the optimization executes without crashing, but not checking its result so that the reader is not confused by the lack of an assertion.